crates/symbolicator/src/services/download/s3.rs

```diff
@@ -88,28 +91,51 @@ impl S3Downloader {
         }
     }
 
-    fn get_s3_client(&self, key: &Arc<S3SourceKey>) -> Arc<rusoto_s3::S3Client> {
+    fn get_s3_client(
+        &self,
+        aws_credentials_provider: &str,
+        key: &Arc<S3SourceKey>,
+    ) -> Arc<rusoto_s3::S3Client> {
         let mut container = self.client_cache.lock();
         if let Some(client) = container.get(&*key) {
             metric!(counter("source.s3.client.cached") += 1);
             client.clone()
         } else {
             metric!(counter("source.s3.client.create") += 1);
 
-            let s3 = Arc::new(rusoto_s3::S3Client::new_with(
-                self.http_client.clone(),
-                rusoto_credential::StaticProvider::new_minimal(
-                    key.access_key.clone(),
-                    key.secret_key.clone(),
-                ),
-                key.region.clone(),
-            ));
+            let region = key.region.clone();
+            log::debug!(
+                "Using AWS credentials provider: {:?}",
+                aws_credentials_provider
+            );
+            let s3 = Arc::new(match aws_credentials_provider {
+                "container" => {
+                    let provider = rusoto_credential::ContainerProvider::new();
+                    self.create_s3_client(provider, region)
+                }
+                "static" => {
```

Comment:
IMO, this should just be the default branch, that way we would log an invalid access key error somewhere down the road. That should be more sensible than to panic.

Reply:
Done!

Reply:
oh, i realised i just made a comment that differs from this and now realise why you did this. 😄   I think my new comment still makes sense now that we have an enum, and this comment made sense if you would be matching strings.  But enums are a better choice here in my opinion, so exhaustive matching is preferred.